lib/src/layer/attribution_layer.dart

```diff
@@ -66,3 +85,382 @@ class AttributionWidget extends StatelessWidget {
+  final Widget? closeIcon;
+
+  /// The color to use as the popup box's background color, defaulting to the
+  /// [Theme]s background color
+  final Color? popupBackgroundColor;
+
+  /// The radius of the edges of the popup box
+  final BorderRadius? popupBorderRadius;
+
+  /// If not [Duration.zero] (default), the popup box will be open by default and
+  /// hidden this long after the map is initialised
+  ///
+  /// This is useful with certain sources/tile servers that make immediate
+  /// attribution mandatory and are not attributed with a permanently visible
+  /// [LogoSourceAttribution].
+  final Duration popupInitialDisplayDuration;
+
+  /// Whether to add an additional attribution logo and text for 'flutter_map'
+  final bool showFlutterMapAttribution;
+
+  /// The curve to use when toggling the visibility of the popup box and the
+  /// state of the open/close icons
+  final Curve animationCurve;
+
+  /// The length of the animation that toggles the visibility of the popup box
+  /// and the state of the open/close icons
+  final Duration animationDuration;
+
+  /// A prebuilt attribution layer that supports both logos and text through
+  /// [SourceAttribution]s
+  ///
+  /// [TextSourceAttribution]s are shown in a popup box (toggled by a tap/click
+  /// on the [openIcon]/[closeIcon]), unlike [LogoSourceAttribution], which are
+  /// visible permenantly adjacent to the open/close icon.
+  ///
+  /// The popup box also closes automatically on any interaction with the map.
+  ///
+  /// Shows a 'flutter_map' attribution logo and text, unless
+  /// [showFlutterMapAttribution] is `false`.
+  ///
+  /// Can be further customized to a certain extent through the other properties.
+  const RichAttributionWidget({
+    super.key,
+    required this.attributions,
+    this.alignment = AttributionAlignment.bottomRight,
+    this.openIcon,
+    this.closeIcon,
+    this.popupBackgroundColor,
+    this.popupBorderRadius,
+    this.popupInitialDisplayDuration = Duration.zero,
+    this.showFlutterMapAttribution = true,
+    this.animationCurve = Curves.easeInOut,
+    this.animationDuration = const Duration(milliseconds: 150),
+  });
+
+  @override
+  State<StatefulWidget> createState() => RichAttributionWidgetState();
+}
+
+class RichAttributionWidgetState extends State<RichAttributionWidget> {
+  StreamSubscription<MapEvent>? mapEventSubscription;
+
+  final persistentAttributionKey = GlobalKey();
+  Size? persistentAttributionSize;
+
+  late bool popupExpanded = widget.popupInitialDisplayDuration != Duration.zero;
+  bool persistentHovered = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (widget.popupInitialDisplayDuration != Duration.zero) {
+      Future.delayed(
+        widget.popupInitialDisplayDuration,
+        () => setState(() => popupExpanded = false),
+      );
+    }
+
+    WidgetsBinding.instance.addPostFrameCallback(
+      (_) => WidgetsBinding.instance.addPostFrameCallback(
```

Comment:
I'm not entirely sure, but delaying one frame doesn't get the right size. I'm assuming it's because FM takes an additional frame in the first place, but I'm not sure.